lib/internal/Magento/Framework/Mail/Message.php

```diff
@@ -7,14 +7,25 @@
  */
 namespace Magento\Framework\Mail;
 
-class Message extends \Zend_Mail implements MessageInterface
+use Zend\Mime\Mime;
+use Zend\Mime\Part;
+
+/**
+ * @todo composition instead of inheritance for better testability
+ * - add a ZendMailDecorator interface with getZendMail() method for usage in \Magento\Framework\Mail\Transport
+ * @todo get rid of temporal coupling (setMessageType() + setBody())
+ * - deprecate setMessageType(), implement a HtmlMessage decorator instead
```

Comment:
The problem with the current implementation is that you have to call `setMessageType()` before `setBody()`. I'm now considering changing the `MessageInterface` interface. As it should be used via `TransportBuilder`, it should not have serious impact.

- The current signatures `setBody(mixed $body)` and `mixed getBody()` are not meaningful. Either "mixed" must be replaced with "string" or we are leaking the zend abstraction.
- The interface does not allow multitype mails (this is the needed case not covered. Something that Magento always has been lacking of)

One idea is, replacing `setMessageType()`, `setBody()` and `getBody()` with:

- `setBodyHtml(string $html)`
- `setBodyText(string $text)`
- `string getBodyText()`
- `string getBodyHtml()`

This allows text mails, HTML mails, and multitype mails, does not require calling them in any specific order and does not reveal anything of the underlying Zend mail component.